File: p31.hs
denom = [1,2,5,10,20,50,100,200,400]

counts = reverse [0 : reverse (takeWhile (< 200) (scanl1 (+) (repeat x))) | x <- denom]


result [] total = 0
result _ 0 = 1
result (x:xs) total = result xs (total-x)

ret = [[a,b,c,d,e,f,g,h] | a <- (counts !! 0), b <- (counts !! 1), c <- (counts !! 2), d <- (counts !! 3), e <- (counts !! 4), f <- (counts !! 5), g <- (counts !! 6), h <- (counts !! 7), a + b + c + d + e + f + g + h == 200]

rret = result [1,2,5,10,20,50,100,200] 200

File: p31.py
def C(i,J):
    if len(J) == 0:
        return 0
    ret = 0
    
    print "i %s J %s " % (i, J)

    for j in range(1, i / J[0]):
        print " (J[0] * j) = %s",  (J[0] * j)
        ret +=  (J[0] * j) * C(i - J[0] * j, J[1:])
        


    return ret

print C(200, [1,2,5,10,20,50,100,200])
    


#print go([1,2],0)
#print go([1,2,5,10,20,50,100,200])

    
